tests/serde.rs

```diff
@@ -409,3 +409,34 @@ fn maps() {
             assert_json(lhs_kv, rhs_kv);
         });
 }
+
+#[wasm_bindgen_test]
+fn maps_string_object() {
+    // #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+    // struct Struct<A, B> {
+    //     a: A,
+    //     b: B,
+    // }
+
+    let mut serializer = Serializer::new();
+    serializer.serialize_maps_as_objects(true);
+
+    let mut src = HashMap::new();
+    // src.insert("a", Struct {a: 2, b: "S"})
+    // src.insert("a", Struct {a: 2, b: "S"})
+    src.insert("a".to_string(), 123);
+    src.insert("b".to_string(), 123);
+
+    let res = src.serialize(&serializer).unwrap();
+
+    let res = res.dyn_into::<js_sys::Object>().unwrap();
+    assert_eq!(Object::entries(&res).length() as usize, src.len());
+
+    Object::entries(&res)
```

Comment:
Why not just compare objects as whole?

Reply:
Also: please add a test and document behaviour for what happens when a key is not a string.

Reply:
How can this (=whether a generic type parameter is a specific type) be detected? I tried this but that didn't work

```rust
fn serialize_key<T: ?Sized + Serialize>(&mut self, key: &T) -> Result<()> {
    debug_assert!(self.next_key.is_none());
    let key_any = key as &dyn Any;
    if !key_any.is::<&str>() || key_any.is::<String>() {
       return Err(Error::custom(
           "Map key is not a string and cannot be an object key",
       ));
    }

```
because `T` isn't always `Sized`:
```
error[E0277]: the size for values of type `T` cannot be known at compilation time
   --> src/ser.rs:163:27
    |
159 |     fn serialize_key<T: ?Sized + Serialize>(&mut self, key: &T) -> Result<()> {
    |                      - this type parameter needs to be `Sized`
...
163 |             let key_any = key as &dyn Any;
    |                           ^^^ doesn't have a size known at compile-time
    |
    = note: required for the cast to the object type `dyn Any`
```

Reply:
Yeah it's tricky to check the Rust type - you could do that by implementing custom `ser::Serializer` that only permits strings through - but it's much more easily to check the JS value type.

That is, you can do

```rust
let next_key = key.serialize(self.serializer)?;
if !next_key.is_string() { ... }
self.next_key = Some(next_key);
``